tests/scripts/all.sh

```diff
@@ -2989,11 +2989,17 @@ component_test_make_shared () {
 
 component_test_cmake_shared () {
     msg "build/test: cmake shared" # ~ 2min
+    root_dir="$(pwd)"
     cmake -DUSE_SHARED_MBEDTLS_LIBRARY=On .
     make
     ldd programs/util/strerror | grep libmbedcrypto
     make test
     programs/test/dlopen_demo.sh
+    if [[ "$OSTYPE" == linux* ]]; then
```

Comment:
So why don't we need something like this in the original PR? If it's just because it so happens that we don't run that test on non-Linux at the moment, we are setting ourselves up for a regression later

Reply:
Good question, and that seems right AFAICT. I can port this logic onto 3.6 and development branches if you'd like?

Reply:
Yes please, thanks

Reply:
To make a life a little simpler, if I could get the rest of the open PRs pulled and then I can backport this logic.